src/components/Button.js

```diff
@@ -17,11 +18,15 @@ type Props = {
   onChange: ?(event: HardwareEvent) => any,
   onDown: ?(event: HardwareEvent) => any,
   onUp: ?(event: HardwareEvent) => any,
+  internalPullup: ?boolean,
 };
 
 class Button extends React.Component<Props> {
-  onRead = (value: number) => {
-    const {onDown, onUp, onChange} = this.props;
+  onRead = (observedValue: number) => {
+    const {onDown, onUp, onChange, internalPullup} = this.props;
+
+    const value = internalPullup ? +!observedValue : observedValue;
```

Comment:
Is this a tricky way to make `observedValue` flip between 1 and 0?

Reply:
Yes it is! I looked for a few ways to do this, but I couldn't find anything that would clearly specify intent. 

My first implementation was done with modulus division, but it wasn't exactly easy to understand. I don't think this way is either, but I'm not sure of a way to do it in JS that is not massive but still readable by developers. 

The base intent was to make the internal vs external resistor selection a binary operation that required no code changes by the implementer. I'm open to suggestions on how to make it more readable or to do something entirely different.